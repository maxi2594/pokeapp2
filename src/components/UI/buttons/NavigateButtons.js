import React from 'react'

export const NavigateButtons = ({
    handleSetUrL,
    next,
    previous
}) => {


    const styles = {
        container: {
            display: 'flex',
            alignItems : 'center',
            backgroundColor: 'rgb(43,47,51)',
            justifyContent: 'space-between',
            flexDirection: 'row',
            marginTop: '1rem',
            marginBottom: '2rem',
        },
        button:{
            border: '2px solid grey',
            backgroundColor: 'transparent',
            marginLeft: '10px',
            marginRight: '10px',
            color: 'grey',
            width: '5rem',
            height: '2rem',
        }
    }
    return (
        <div style={ styles.container }>

            <button 
                style={ styles.button }
                onClick={ ()=> handleSetUrL(previous) }
            > 
                PREVIOS 
            </button>

            <button 
                style={ styles.button }
                onClick={ ()=> handleSetUrL(next) }
            >
                NEXT 
            </button>

        </div>
    )
}
