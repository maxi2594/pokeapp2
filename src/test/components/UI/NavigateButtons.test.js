import { shallow } from 'enzyme';
import React from 'react'
import { NavigateButtons } from '../../../components/UI/buttons/NavigateButtons';

describe('Test in <NavigateButtons/>', () => {
    
    const componentArgs = {
        
        handleSetUrL : jest.fn(),
        next : 'next',
        previous : 'previous',
    }

    const wrapper = shallow( <NavigateButtons {...componentArgs }/>)

    test('should be shown propperly', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('should include 2 buttons', () => {
        
        expect( wrapper.find('button').length ).toEqual( 2 )
    })

    test('should work handleSetUrL with next & previos', () => {
        
        const bottonPrevious = wrapper.find('button').at(0)
        const bottonNext = wrapper.find('button').at(1)

        bottonPrevious.simulate('click');
        expect(componentArgs.handleSetUrL)
            .toHaveBeenCalledWith(componentArgs.previous)

        bottonNext.simulate('click');
        expect(componentArgs.handleSetUrL)
            .toHaveBeenCalledWith(componentArgs.next)
    })

    
})
