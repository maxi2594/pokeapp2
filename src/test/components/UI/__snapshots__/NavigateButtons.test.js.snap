// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <NavigateButtons/> should be shown propperly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(43,47,51)",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": "2rem",
      "marginTop": "1rem",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "2px solid grey",
        "color": "grey",
        "height": "2rem",
        "marginLeft": "10px",
        "marginRight": "10px",
        "width": "5rem",
      }
    }
  >
    PREVIOS
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "2px solid grey",
        "color": "grey",
        "height": "2rem",
        "marginLeft": "10px",
        "marginRight": "10px",
        "width": "5rem",
      }
    }
  >
    NEXT
  </button>
</div>
`;
